pandas/tests/indexes/test_multi.py

```diff
@@ -827,6 +837,16 @@ def test_from_product(self):
         tm.assert_index_equal(result, expected)
         assert result.names == names
 
+        # iterator as input
```

Comment:
can you make this a separate test (the added code), and do so for from_product, from_tuples, from_arrays as well, call it something like

``test_multiindex_from_product_iterable`` and add the issue number to it.